client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java

```diff
@@ -306,6 +310,19 @@ public SendResult sendMessage(//
         final SendMessageContext context, // 11
         final DefaultMQProducerImpl producer // 12
     ) throws RemotingException, MQBrokerException, InterruptedException {
+
+        byte[] msgBody = msg.getBody();
+        Integer sysFlag = requestHeader.getSysFlag();
+        if (null != sysFlag && ((sysFlag & MessageSysFlag.COMPRESSED_FLAG) == MessageSysFlag.COMPRESSED_FLAG)) {
+            try {
+                msgBody = UtilAll.compress(msgBody, zipCompressLevel);
+            } catch (IOException e) {
+                requestHeader.setSysFlag(MessageSysFlag.clearCompressedFlag(sysFlag));
+                log.error("tryToCompressMessage exception", e);
```

Comment:
Got your point.  

1. The code is moved here, and same log level should be applied IMO. @vongosling @vintagewang  any particular reason behind this?